pkg/cloudproviders/azure/metadata.go

```diff
@@ -79,5 +86,41 @@ func GetMetadata(ctx context.Context) (*storage.ProviderMetadata, error) {
 			},
 		},
 		Verified: verified,
+		Cluster:  clusterMetadata,
 	}, nil
 }
+
+func getClusterMetadata(ctx context.Context) *storage.ClusterMetadata {
+	k8sClient, err := k8sutil.GetK8sInClusterClient()
```

Comment:
There is a function in sensor called
```
func mustCreateK8sClient(config *rest.Config) kubernetes.Interface {
```
Let's either refactor that or utilize that function by passing it to the function